### PR TITLE
FTP active/passive mode file extraction tests

### PR DESCRIPTION
**Pull request by jlucovsky**
This changeset adds test cases for FTP active and passive mode using
the pcaps from issue [2527](https://redmine.openinfosecfoundation.org/issues/2527)

This PR verifies functionality added in https://github.com/OISF/suricata/pull/3758.

### CODE REVIEW

File: tests/filestore-ftp-active-mode/ftp.rule [hunk 1]
```diff
@@ -0,0 +1 @@
+alert tcp any 20 -> any any (msg:"File Found within FTP and stored"; filestore;  sid:1; rev:1;)
```
Comment (victorjulien) [inline tests/filestore-ftp-active-mode/ftp.rule]: if unused it can be removed

File: tests/filestore-ftp-active-mode/suricata.yaml [hunk 2]
```diff
@@ -0,0 +1,12 @@
+%YAML 1.1
+---
+
+include: ../../etc/suricata-4.0.3.yaml
```
Comment (victorjulien) [inline tests/filestore-ftp-active-mode/suricata.yaml]: shouldn't be needed

**Comment by victorjulien** [comment]
Can you also make sure the pcaps only contain the ftp traffic we care about?

**Comment by victorjulien** [comment]
These will also need: 'requires' to make sure we only run this on 5.0.0 for passive, 4.1.x+ for active (or other way around).

Next I think you need to pass
```
args:
  - --runmode=single
```
as the pcap processing of the control flow and data flow can be so fast that there is a race between them.

File: tests/filestore-ftp-active-mode/suricata.yaml (hunk 2 above)
Comment (jlucovsky) [reply tests/filestore-ftp-active-mode/suricata.yaml]: Will remove.

File: tests/filestore-ftp-active-mode/ftp.rule (hunk 1 above)
Comment (jlucovsky) [reply tests/filestore-ftp-active-mode/ftp.rule]: Will remove.

**Comment by jlucovsky** [comment]
> I'd prefer file-store v2 tests, could you update the tests?

Are you asking for the test to be renamed or something else?

**Comment by victorjulien** [comment]
> > I'd prefer file-store v2 tests, could you update the tests?
> 
> Are you asking for the test to be renamed or something else?

No, the file store output comes in 2 variants. An older one (v1) and a newer one (v2). I would prefer the test to use v2. https://github.com/OISF/suricata/blob/master/suricata.yaml.in#L460

**Comment by jlucovsky** [comment]
> > > I'd prefer file-store v2 tests, could you update the tests?
> > 
> > 
> > Are you asking for the test to be renamed or something else?
> 
> No, the file store output comes in 2 variants. An older one (v1) and a newer one (v2). I would prefer the test to use v2. https://github.com/OISF/suricata/blob/master/suricata.yaml.in#L460

Will ensure v2 is used.

**Comment by jlucovsky** [comment]
> Can you also make sure the pcaps only contain the ftp traffic we care about?

Will update both pcaps.

**Comment by jlucovsky** [comment]
Continued in #69

**Pull request closed without merging**